pkg/controller/plan/adapter/ocp/adapter.go

```diff
@@ -29,7 +35,17 @@ func (r *Adapter) Validator(plan *api.Plan) (validator base.Validator, err error
 		return
 	}
 
-	v := &Validator{plan: plan, client: client}
+	err = client.Get(context.TODO(), k8sclient.ObjectKey{Namespace: ref.Namespace, Name: ref.Name}, secret)
```

Comment:
do we have a secret when the source is the "host" provider?

Reply:
we can pass nil, but this going to change

Reply:
ack